### PR TITLE
backport: Merge bitcoin#19370, 19562, 19538, 19272, 19214, 19429, 18990, 20927, 20507

### PR DESCRIPTION
**Pull request by vijaydasmp**
Bitcoin Backports

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6  , For some reason, GitLab ci is not triggering

**Comment by UdjinM6** [comment]
Pls rebase. This should hopefully help triggering CI.

File: src/net_processing.cpp [hunk 1]
```diff
@@ -3272,7 +3272,8 @@ void PeerManagerImpl::ProcessMessage(
         vRecv >> vInv;
         if (vInv.size() > MAX_INV_SZ)
         {
-            Misbehaving(pfrom.GetId(), 20, strprintf("message inv size() = %u", vInv.size()));
+            LOCK(cs_main);
```
Comment (UdjinM6) [inline src/net_processing.cpp:3278]: 19272:
```suggestion
```

File: test/functional/p2p_invalid_messages.py [hunk 2]
```diff
@@ -52,11 +54,13 @@ def run_test(self):
         self.test_checksum()
         self.test_size()
         self.test_msgtype()
-        self.test_large_inv()
         self.test_resource_exhaustion()
+        self.test_oversized_inv_msg()
+        self.test_oversized_getdata_msg()
+        self.test_oversized_headers_msg()
```
Comment (UdjinM6) [inline test/functional/p2p_invalid_messages.py:60]: 19272:
```suggestion
        self.test_oversized_inv_msg()
        self.test_oversized_getdata_msg()
        self.test_oversized_headers_msg()
        self.test_resource_exhaustion()
```

File: test/functional/p2p_invalid_messages.py [hunk 3]
```diff
@@ -94,19 +100,21 @@ def test_checksum(self):
...
             conn.send_raw_message(msg)
             conn.sync_with_ping(timeout=1)
-            self.nodes[0].disconnect_p2ps()
+        self.nodes[0].disconnect_p2ps()
 
     def test_size(self):
+        self.log.info("Test message with oversized payload disconnects peer")
         conn = self.nodes[0].add_p2p_connection(P2PDataStore())
         with self.nodes[0].assert_debug_log(['']):
             # Create a message with oversized payload
```
Comment (UdjinM6) [inline test/functional/p2p_invalid_messages.py:109]: 19272:
```suggestion
```

File: test/functional/p2p_invalid_messages.py [hunk 4]
```diff
@@ -153,6 +148,25 @@ def test_resource_exhaustion(self):
...
+        self.test_oversized_msg(msg_inv([CInv(MSG_TX, 1)] * size), size)
+
+    def test_oversized_getdata_msg(self):
+        size = MAX_INV_SIZE + 1
+        self.test_oversized_msg(msg_getdata([CInv(MSG_TX, 1)] * size), size)
+
+    def test_oversized_headers_msg(self):
+        size = MAX_HEADERS_RESULTS + 1
+        self.test_oversized_msg(msg_headers([CBlockHeader()] * size), size)
+
```
Comment (UdjinM6) [inline test/functional/p2p_invalid_messages.py:169]: 19272: these should go above `def test_resource_exhaustion`

also missing changes in `test_resource_exhaustion` and missing corresponding constants at the top of the test too

File: test/functional/wallet_encryption.py [hunk 5]
```diff
@@ -70,20 +71,25 @@ def run_test(self):
...
         actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
         assert_greater_than_or_equal(actual_time, expected_time)
-        assert_greater_than(expected_time + 5, actual_time) # 5 second buffer
-        # Check a time greater than the limit
+        assert_greater_than(expected_time_with_buffer, actual_time)
+
+        self.log.info('Check a timeout greater than the limit')
         expected_time = self.mocktime + MAX_VALUE - 1
         self.nodes[0].walletpassphrase(passphrase2, MAX_VALUE + 1000)
+        expected_time_with_buffer = time.time() + MAX_VALUE
```
Comment (UdjinM6) [inline test/functional/wallet_encryption.py:88]: 19429:
```suggestion
        expected_time_with_buffer = self.mocktime + MAX_VALUE
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/p2p_invalid_messages.py (hunk 4 above)
Comment (vijaydasmp) [reply test/functional/p2p_invalid_messages.py:169]: Thanks , checking it

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 apparently unrelated in **feature_llmq_simplepose.py**", line 203, in repair_masternodes, causing test failure, 
https://gitlab.com/dashpay/dash/-/jobs/6096975858

**Comment by knst** [comment]
> Hello @UdjinM6 apparently unrelated in feature_llmq_simplepose.py", line 203, 

Please ignore, seems as failure unrelated, see https://github.com/dashpay/dash/pull/5859#issuecomment-1925457843

File: test/functional/test_framework/messages.py [hunk 6]
```diff
@@ -46,6 +46,10 @@
 
 BIP125_SEQUENCE_NUMBER = 0xfffffffd  # Sequence number that is BIP 125 opt-in and BIP 68-opt-out
 
+MAX_PROTOCOL_MESSAGE_LENGTH = 4000000  # Maximum length of incoming protocol messages
```
Comment (UdjinM6) [inline test/functional/test_framework/messages.py:49]: 19272:
```suggestion
MAX_PROTOCOL_MESSAGE_LENGTH = 3000000  # Maximum length of incoming protocol messages
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/wallet_encryption.py (hunk 5 above)
Comment (UdjinM6) [reply test/functional/wallet_encryption.py]: hmm... this wasn't enough. pls check 3e2265e4ba

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , requesting review

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merge via merge commit

**Pull request merged**